src/server/network/server.cpp

```diff
@@ -8,57 +8,76 @@
 
 #include "../world/terrain_generation.h"
 
-Server::Server(const ServerConfig &config)
+Server::Server(const ServerConfig& config)
     : NetworkHost("Server")
-    ,   m_worldSize (config.worldSize)
-    ,   m_worldHeight (config.worldHeight)
+    , m_worldSize(config.worldSize)
+    , m_worldHeight(config.worldHeight)
 {
     for (int z = 0; z < m_worldSize; z++) {
         for (int x = 0; x < m_worldSize; x++) {
-            std::array<int, CHUNK_AREA> heightMap = makeHeightMap({x, 0, z});
+            std::array<int, CHUNK_AREA> heightMap = makeHeightMap({ x, 0, z });
 
             for (int y = 0; y < m_worldHeight; y++) {
-                Chunk &chunk = m_world.chunks.addChunk({x, y, z});
+                Chunk& chunk = m_world.chunks.addChunk({ x, y, z });
                 //makeFlatTerrain(&chunk);
                 makeNaturalTerrain(&chunk, heightMap, m_worldSize);
             }
         }
     }
 }
 
-void Server::sendChunk(peer_id_t peerId, const ChunkPosition &position)
+void Server::sendChunk(peer_id_t peerId, const ChunkPosition& position)
 {
     if (!m_connectedClients[peerId].connected) {
         return;
     }
 
-    const Chunk &chunk = [this, &position]() {
+    const Chunk& chunk = [this, &position]() {
         if (m_world.chunks.hasChunk(position)) {
             return m_world.chunks.getChunk(position);
         }
         else {
-            Chunk &c = m_world.chunks.addChunk(position);
+            Chunk& c = m_world.chunks.addChunk(position);
             return c;
         }
     }();
 
     // Create the chunk-data packet
     sf::Packet packet;
     packet << ClientCommand::ChunkData << chunk.getPosition().x
-           << chunk.getPosition().y << chunk.getPosition().z;
+        << chunk.getPosition().y << chunk.getPosition().z;
 
     auto compressedChunk = chunk.compress();
     packet << static_cast<u32>(compressedChunk.size());
-    for (auto &block : compressedChunk) {
+    for (auto& block : compressedChunk) {
         packet << block.first << block.second;
     }
 
     // Send chunk data to client
     sendToPeer(m_connectedClients[peerId].peer, packet, 1,
-               ENET_PACKET_FLAG_RELIABLE);
+        ENET_PACKET_FLAG_RELIABLE);
 }
 
-void Server::onPeerConnect(ENetPeer *peer)
+void Server::sendPlayerSkin(peer_id_t peerId, peer_id_t toPeer)
+{
+    LOGVAR("Server", "Sending skin of peer:", (int)peerId);
+    sf::Packet skinPacket;
+    skinPacket << ClientCommand::NewPlayerSkin;
+    skinPacket << peerId;
+
+    for (int i = 0; i < (1024 * 8); i++)
+        skinPacket << m_entities[peerId].m_skinData[i];
+
```

Comment:
As mentioned earlier, I think we can use append here 😄 

https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1Packet.php#a7dd6e429b87520008326c4d71f1cf011